lib/open_trip_planner_client/itinerary_group.ex

```diff
@@ -39,11 +43,49 @@ defmodule OpenTripPlannerClient.ItineraryGroup do
   """
   @spec groups_from_itineraries([Itinerary.t()], Keyword.t()) :: [%__MODULE__{}]
   def groups_from_itineraries(itineraries, opts \\ []) do
+    ideal_itineraries = opts |> Keyword.get(:ideal_itineraries, [])
+
+    available_groups =
+      itineraries
+      |> to_groups(opts)
+      |> Enum.take(Keyword.get(opts, :num_groups, @num_groups))
+
+    unavailable_groups =
+      ideal_itineraries
+      |> filter_unavailable_itineraries(available_groups)
```

Comment:
suggestion: what if we pass in _all_ the available groups here, rather than the first-N-groups? I feel like this might make the result more relevant somehow. 
though I'm also curious: would we want to run `to_groups/2` on each set of itinerary lists first, and _then_ compare the groups to each other?

(forgive me if we already tried this!)

Reply:
> suggestion: what if we pass in all the available groups here, rather than the first-N-groups? I feel like this might make the result more relevant somehow.

Ugggh - I think it was originally doing that, but then I goofed a refactor.

> though I'm also curious: would we want to run to_groups/2 on each set of itinerary lists first, and then compare the groups to each other?

Hummm I think it _should_ be equivalent, and I thought I had a good reason for doing it this way, but I can't remember what that good reason was.

> (forgive me if we already tried this!)

We didn't already try this, so I can't forgive you, sorry!